src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -892,17 +921,37 @@ private StableReverseOrderListView(List<E> base) {
             // This method does not evaluate the elements
             @Override
             public String toString() {
-                final StableValueImpl<E>[] delegates = ((StableList<E>)base).delegates;
+                final StableValueImpl<E>[] delegates = deepRoot(base).delegates;
+                // Todo: Provide a copy free solution
                 final StableValueImpl<E>[] reversed = ArraysSupport.reverse(
                         Arrays.copyOf(delegates, delegates.length));
-                return StableUtil.renderElements(base, "Collection", reversed);
+                return StableUtil.renderElements(this, "Collection", reversed);
             }
 
             @Override
             public List<E> reversed() {
                 return base;
             }
 
+            @Override
+            public List<E> subList(int fromIndex, int toIndex) {
+                final int size = base.size();
+                subListRangeCheck(fromIndex, toIndex, size);
+                return new StableReverseOrderListView<>(base.subList(size - toIndex, size - fromIndex));
+            }
+
+        }
+
+        static <E> StableList<E> deepRoot(List<E> list) {
+            // Avoid spinning code for a pattern matching switch, instead use an if rake
+            if (list instanceof StableList<E> sl) {
+                return sl;
+            } else if (list instanceof StableList.StableReverseOrderListView<E> rev) {
+                return deepRoot(rev.base);
+            } else if (list instanceof StableList.StableSubList<E> sub) {
+                return deepRoot(sub.root);
+            }
+            throw new InternalError("Should not reach here: " + list.getClass().getName());
```

Comment:
Does this need to be recursive? There is logic to avoid nesting of sublists, and a sublist of a reversed view should end up being a reversed view of a sublist, so the most deep nesting that can occur is reversed(sublist(base)).

Reply:
Actually, this introduces a bug. If there is a reversed(sublist(base)) op, then the sublist is disregarded. So we need to remove this.